src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -65,39 +67,60 @@ public EcheladderScreen()
 		guiWidth = 250;
 		guiHeight = 202;
 		
-		maxScroll = (ClientRungData.getFinalRungIndex() + 1) * RUNG_Y - 154;
+		maxScroll = ClientRungData.getFinalRungIndex() - 10;
 	}
 	
 	@Override
 	public void init()
 	{
 		super.init();
-		scrollIndex = Mth.clamp((ClientPlayerData.getRung() - 8) * RUNG_Y, 0, maxScroll);
 		animatedRung = Math.max(animatedRung, lastRung);    //If you gain a rung while the gui is open, the animated rung might get higher than the lastRung. Otherwise they're always the same value.
 		fromRung = lastRung;
 		lastRung = ClientPlayerData.getRung();
+		
+		rungBars.clear();
+		for(int i = 0; i < ClientRungData.getFinalRungIndex() + 1; i++)
+		{
+			Component name = Component.translatable(I18n.exists("echeladder.rung." + i) ? I18n.get("echeladder.rung." + i) : "Rung " + (i + 1));
+			
+			Optional<String> tooltip = Optional.empty();
+			String description = ClientRungData.getData(i).description().orElse("");
+			if(!description.isEmpty())
+				tooltip = Optional.of(description);
+			
+			RungBar rungBar = new RungBar(xOffset + 90, yOffset + 175 - i * RUNG_Y, 146, RUNG_Y, name, tooltip, i);
+			rungBars.add(rungBar);
+			addRenderableWidget(rungBar);
+			
+			rungBar.visible = i <= VISIBLE_RUNG_COUNT;
+		}
 	}
 	
 	@Override
 	public void render(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTicks)
 	{
-		updateScrollAndAnimation(mouseY);
+		updateAnimation();
 		
 		int speedFactor = MinestuckConfig.CLIENT.echeladderAnimation.get().getSpeed();
 		
-		int currentRung = calculateRungAnimationStep(speedFactor);
+		calculateRungAnimationStep(speedFactor);
 		
 		super.render(guiGraphics, mouseX, mouseY, partialTicks);
 		
 		drawTabs(guiGraphics);
 		
-		drawLadder(guiGraphics, currentRung);
+		rungBars.forEach(rungBar -> {
+			rungBar.setY(rungBar.initY + getScrollMod());
+			rungBar.updateVisibility();
+		});
```

Comment:
Better to do this before the `super.render()` call, so that they're always up to date when they render.